bgpd/bgp_orr.c

```diff
@@ -0,0 +1,1163 @@
+/*
+ * BGP Optimal Route Reflection
+ * Copyright (C) 2021  Samsung R&D Institute India - Bangalore.
+ *			Madhurilatha Kuruganti
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "bgpd/bgpd.h"
+#include "bgpd/bgp_debug.h"
+#include "bgpd/bgp_orr.h"
+#include "bgpd/bgp_vty.h"
+#include "zclient.h"
+
+DEFINE_MTYPE_STATIC(BGPD, ORR_IGP_INFO, "ORR IGP Metric info");
+
+extern struct zclient *zclient;
+
+static inline bool is_orr_primary_root(struct bgp_orr_group *orr_group,
+				       char *host)
+{
+	return orr_group->primary && !strcmp(orr_group->primary->host, host);
+}
+
+static inline bool is_orr_secondary_root(struct bgp_orr_group *orr_group,
+					 char *host)
+{
+	return orr_group->secondary &&
+	       !strcmp(orr_group->secondary->host, host);
+}
+
+static inline bool is_orr_tertiary_root(struct bgp_orr_group *orr_group,
+					char *host)
+{
+	return orr_group->tertiary && !strcmp(orr_group->tertiary->host, host);
+}
+
+static inline bool is_orr_active_root(struct bgp_orr_group *orr_group,
+				      char *host)
+{
+	return orr_group->active && !strcmp(orr_group->active->host, host);
+}
+static inline bool is_orr_root_node(struct bgp_orr_group *orr_group, char *host)
+{
+	return is_orr_primary_root(orr_group, host) ||
+	       is_orr_secondary_root(orr_group, host) ||
+	       is_orr_tertiary_root(orr_group, host);
+}
+
+static inline bool is_orr_primary_reachable(struct bgp_orr_group *orr_group)
```

Comment:
I wonder the same thing here -- is there any way to do this in a single function? It seems like it might make the caller's life easier?